CadRevealComposer/Operations/SectorSplitting/PrioritySectorSplitter.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace CadRevealComposer.Operations.SectorSplitting;
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using IdProviders;
+using Primitives;
+
+public class PrioritySectorSplitter : ISectorSplitter
+{
+    private const long SectorEstimatedByteSizeBudget = 50_000; // bytes, Arbitrary value
+
+    public IEnumerable<InternalSector> SplitIntoSectors(
+        APrimitive[] allGeometries,
+        SequentialIdGenerator sectorIdGenerator
+    )
+    {
+        if (sectorIdGenerator.PeekNextId == 0)
+            _ = sectorIdGenerator.GetNextId(); // Get and discard id 0, as 0 is hardcoded below
+        var dummyRootSector = SplittingUtils.CreateRootSector(0, "/0", new BoundingBox(Vector3.Zero, Vector3.One));
+        yield return dummyRootSector; // Using dummy root sector because we need to remap it to the non-priority sector root later on
+
+        var sectors = new List<InternalSector>();
+        // We split the geometries into sectors based on the discipline because there are very few highlight cases that are cross-discipline
+        // This helps reduce the overhead of a priority sector, since highlighting of multiple tags within the same discipline is more common
+        var primitivesGroupedByDiscipline = allGeometries.GroupBy(x => x.Discipline);
+        foreach (var disciplineGroup in primitivesGroupedByDiscipline)
+        {
+            var geometryGroups = disciplineGroup.GroupBy(primitive => primitive.TreeIndex); // Group by treeindex to avoid having one treeindex unnecessary many sectors
+            var nodes = PrioritySplittingUtils.ConvertPrimitiveGroupsToNodes(geometryGroups);
+
+            // Ignore outlier nodes
+            // TODO: Decide if this is the right thing to do
+            (Node[] regularNodes, _) = nodes.SplitNodesIntoRegularAndOutlierNodes();
+
+            sectors.AddRange(SplitIntoTreeIndexSectors(regularNodes, dummyRootSector, sectorIdGenerator));
+        }
+
+        foreach (var sector in sectors)
+        {
+            yield return sector with
+            {
+                IsPrioritizedSector = true
+            };
+        }
+    }
+
+    private static IEnumerable<InternalSector> SplitIntoTreeIndexSectors(
+        Node[] nodes,
+        InternalSector rootSector,
+        SequentialIdGenerator sectorIdGenerator
+    )
+    {
+        var nodesProcessed = 0;
+
+        // Sorting by Id as we believe the TreeIndex to group similar parts in the hierarchy together
+        var nodesOrderedByTreeIndex = nodes.OrderBy(x => x.TreeIndex).ToArray();
+
+        while (nodesProcessed < nodes.Length)
+        {
+            // TODO: Should this use spatially aware splitting? Should it place nodes with similar attributes together? Ex: tags?
```

Comment:
We kinda do this by grouping by discipline and ordering the treeindexes?
Change the TODO to something about grouping by tag, if/when tags get sorted out?

Reply:
> We kinda do this by grouping by discipline and ordering the treeindexes?
Correct. its very "rough" as we dont split by file or similar, So a sectors extents may be very large if its just inbetween two area files for example, but its better than nothing.

Reply:
Remove the TODO and create a story regarding "improving grouping of nodes in priority splitting"?